src/fairlens/metrics/unified.py

```diff
@@ -135,60 +133,62 @@ def correlation_matrix(
         cat_cat_metric (Callable[[pd.Series, pd.Series], float], optional):
             The correlation metric used for categorical-categorical series pairs. Defaults to corrected Cramer's V
             statistic.
-        columns_x (Optional[List[str]]):
-            The column names that determine the rows of the matrix.
-        columns_y (Optional[List[str]]):
-            The column names that determine the columns of the matrix.
 
     Returns:
         pd.DataFrame:
             The correlation matrix to be used in heatmap generation.
     """
 
```

Comment:
The correlation matrix is generated using `df.corr()`. Since `df.corr()` only works on numerical data, we need to encode all the columns. The issue with this is that we use the `infer_distr_type()` function to decide which metric would be suitable, which works differently on the encoded numerical data. The only way to resolve this issue is to infer types beforehand (which is probably more efficient). The problem then becomes about making a binary function `(a, b) -> float` that knows the types of `a` and `b` beforehand.

Reply:
Just to add, using `df.corr()` provides a major performance improvement.